### ISSUE TITLE
Endless loop after suspend

### ISSUE THREAD

**Issue by jquorning** [alire-project/alire#921]
Suspend `alr toolchain --select` at the prompt with ctrl-z and resume with `fg` leads to:

```
✗ Not a valid choice, please use a line index.
Please select the gnat version for use with this configuration
```
in loop.

I am using macOS 10.15.

**Comment by mosteo** [comment]
@jquorning , if you could test this commit: https://github.com/mosteo/alire/commit/8e8a9ef284eb7495ddb1c4e7495864922b0d5f80 that would be great. The actual changes are in a submodule so make sure to update these too. Thanks!

**Comment by jquorning** [comment]
@mosteo I tried it out. alr does not resume in a loop. alr ends with error. It is minor problem anyway.

```
~/GitHub/alire_3 ./bin/alr --no-color toolchain --select
Welcome to the toolchain selection assistant

In this assistant you can set up the default toolchain to be used with any crate
that does not specify its own top-level dependency on a version of gnat or
gprbuild.

If you choose "None", Alire will use whatever version is found in the
environment.

Note: gnat is currently not configured. (alr will use the version found in the environment.)

Please select the gnat version for use with this configuration
  1. gnat_native=11.2.3
  2. None
  3. gnat_external=2020.0.0 [Detected at /opt/GNAT/2020/bin/gnat]
  4. gnat_arm_elf=11.2.3
  5. gnat_riscv64_elf=11.2.3
  6. gnat_arm_elf=11.2.2
  7. gnat_native=11.2.2
  8. gnat_riscv64_elf=11.2.2
  9. gnat_arm_elf=11.2.1
  0. gnat_native=11.2.1
  a. (See more choices...)
Enter your choice index (first is default):
> ^Z
zsh: suspended  ./bin/alr --no-color toolchain --select
~/GitHub/alire_3 fg
[2]  - continued  ./bin/alr --no-color toolchain --select
Warning: Unexpected input, exiting.
ERROR: Canceled.
~/GitHub/alire_3
```

**Comment by mosteo** [comment]
Yep, that's the intended behavior for now. I don't really know how to rectify whatever is going on with the input, as I cannot reproduce it on Linux/Windows and I don't have a macOS machine at hand. Thanks for the confirmation.

**Comment by mosteo** [comment]
Fixed in #922

**Comment by mosteo** [comment]
Sorry, that should have been: Fixed in #927
